src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java

```diff
@@ -0,0 +1,122 @@
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.AppPage;
+import teammates.e2e.pageobjects.InstructorCourseEditPage;
+import teammates.e2e.pageobjects.InstructorHomePage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_COURSE_EDIT_PAGE}.
+ */
+public class InstructorCourseEditPageE2ETest extends BaseE2ETestCase {
+    CourseAttributes course;
+    InstructorAttributes[] instructors = new InstructorAttributes[5];
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseEditPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        course = testData.courses.get("InsCrsEdit.CS2104");
+        instructors[0] = testData.instructors.get("InsCrsEdit.helper");
+        instructors[1] = testData.instructors.get("InsCrsEdit.manager");
+        instructors[2] = testData.instructors.get("InsCrsEdit.observer");
+        instructors[3] = testData.instructors.get("InsCrsEdit.coowner");
+        instructors[4] = testData.instructors.get("InsCrsEdit.tutor");
+    }
+
+    @Test
+    public void testAll() {
+        ______TS("verify cannot edit without privilege");
+        // log in as instructor with no edit privilege
+        AppUrl url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[2].googleId)
+                .withCourseId(course.getId());
+        loginAdminToPage(url, InstructorHomePage.class);
+        InstructorCourseEditPage editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseNotEditable();
+        editPage.verifyInstructorsNotEditable();
+        editPage.verifyAddInstructorNotAllowed();
+
+        ______TS("verify loaded data");
+        // re-log in as instructor with edit privilege
+        url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[3].googleId)
+                .withCourseId(course.getId());
+        editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseDetails(course);
+
+        editPage.verifyInstructorDetails(1, instructors[0]);
+        editPage.verifyInstructorDetails(2, instructors[1]);
+        editPage.verifyInstructorDetails(3, instructors[2]);
+        editPage.verifyInstructorDetails(4, instructors[3]);
+        editPage.verifyInstructorDetails(5, instructors[4]);
+
+        ______TS("add instructor");
+        InstructorAttributes newInstructor = InstructorAttributes
+                .builder(course.getId(), "InsCrsEdit.test@gmail.tmt")
+                .withName("Teammates Test")
+                .withIsDisplayedToStudents(true)
+                .withDisplayedName("Instructor")
+                .withRole("Tutor")
+                .build();
+
+        editPage.addInstructor(newInstructor);
+        editPage.verifyStatusMessage("\"The instructor Teammates Test has been added successfully. "
```

Comment:
Why is this status message hardcoded? I think it should be programatically derived. 

If not, the test is brittle and any small change in the message (an extra space etc.) will break the E2E tests.

The same applies to the other status messages.

Reply:
This is actually a grey area. On one hand, what you said is of course true. However, on the other hand, this is a black-box test, meaning that the test code should not actually know that there exists some constant in the code that holds that string.

Either way is acceptable.

Reply:
I did not use the Const values as they are Deprecated. I have added tried to use variables instead for names, emails etc in the status message instead to try to make is slightly less brittle.